File: bitdrift-shop/android/app/src/main/java/ai/bitdrift/shop/OrderSummaryHelper.kt
package ai.bitdrift.shop

import org.json.JSONObject

object OrderSummaryHelper {

    private var _totalFormatted: String? = null

    val totalFormatted: String
        get() = _totalFormatted ?: "$0.00"

    fun formatOrderSummary(apiData: JSONObject?, orderId: String): String {
        val data = apiData ?: return "Order $orderId\nTotal: $totalFormatted\nThank you for your purchase!"
        val txn = data.optString("transaction_id", "")
        val total = data.optDouble("total", 0.0)
        _totalFormatted = "$${String.format("%.2f", total)}"
        val shipping = data.optJSONObject("shipping")
        val delivery = shipping?.optString("estimated_delivery", "") ?: ""
        val tracking = shipping?.optString("tracking_number", "") ?: ""
        return "Order ${data.optString("order_id", orderId)}\nTotal: $totalFormatted\nDelivery: $delivery\nTracking: $tracking\nTxn: ${txn.take(24)}…"
    }

    fun reset() {
        _totalFormatted = null
    }
}

File: bitdrift-shop/ios-clean/BitdriftShop/OrderSummaryHelper.swift
import Foundation

/// Formats the Confirmation screen's order summary, caching the last formatted
/// total. Used only while the crash loop is active — the cached total is what
/// distinguishes the `order_summary` v2 code path from v1 (see
/// `SimulationManager.setVariant`).
enum OrderSummaryHelper {

    private static var cachedTotal: String?

    static var totalFormatted: String { cachedTotal ?? "$0.00" }

    static func formatOrderSummary(_ data: JSON?, orderID: String) -> String {
        guard let data, data.exists else {
            return "Order \(orderID)\nTotal: \(totalFormatted)\nThank you for your purchase!"
        }
        let txn = data.str("transaction_id")
        cachedTotal = money(data.num("total"))
        let shipping = data["shipping"]
        let delivery = shipping.str("estimated_delivery")
        let tracking = shipping.str("tracking_number")
        return """
        Order \(data.str("order_id", orderID))
        Total: \(totalFormatted)
        Delivery: \(delivery)
        Tracking: \(tracking)
        Txn: \(String(txn.prefix(24)))…
        """
    }

    static func reset() {
        cachedTotal = nil
    }
}
